File: TodoAppCore/TodoAppCore/Entities/Todo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace TodoAppCore.Entities
{
    public class Todo
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [Required]
        [MaxLength(50)]
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public EStatus Status { get; set; }
        public AppUser AppUser { get; set; }
        public int UserId { get; set; }
    }

    public enum EStatus
    {
        Done = 0,
        Cancel = 1,
        Doing = 2
    }
}

File: TodoAppClient/src/app/models/todo.ts
export interface Todo{
    id: number;
    name: string;
    description: string;
    createdDate: Date;
    startDate: Date;
    endDate: Date;
    status: Status;
    username: string;
    userId: number;
}

export enum Status{
    Done = 0,
    Cancel = 1,
    Doing = 2
}